app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.kt

```diff
@@ -0,0 +1,175 @@
+package fr.free.nrw.commons.utils
+
+import android.app.Activity
+import android.app.AlertDialog
+import android.app.Dialog
+import android.content.DialogInterface
+import android.view.View
+import fr.free.nrw.commons.R
+import org.apache.commons.lang3.StringUtils
+import timber.log.Timber
+
+object DialogUtil {
+    /**
+     * Shows a dialog safely.
+     * @param activity the activity
+     * @param dialog the dialog to be shown
+     */
+    private fun showSafely(activity: Activity, dialog: Dialog) {
+        val isActivityDestroyed = activity.isDestroyed
+        if (activity.isFinishing || isActivityDestroyed) {
+            Timber.e("Activity is not running. Could not show dialog. ")
+            return
+        }
+        try {
+            dialog.show()
+        } catch (e: IllegalStateException) {
+            Timber.e(e, "Could not show dialog.")
+        }
+    }
+
+    @JvmStatic fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        onPositiveBtnClick: Runnable,
```

Comment:
I don't see any nullability handling in this file, I believe people pass nulls frequently to the listeners, this may cause an illegalStateException